internal/fingerprint/fingerprint.go

```diff
@@ -243,6 +280,57 @@ func (f *Fingerprints) ToFile(ouputFile string) error {
 
 }
 
-func (f *Fingerprints) Append(fingerprint FileFingerprint) {
-	f.Entries = append(f.Entries, fingerprint)
+var filesToUnzip = []string{".jar", ".nupkg"}
+
+func shouldUnzip(filename string) bool {
+	for _, file := range filesToUnzip {
+		if filepath.Ext(filename) == file {
+			return true
+		}
+	}
+
+	return false
+}
+
+func inMemFingerprintingCompressedContent(filename string, exclusions []string) ([]FileFingerprint, error) {
+
+	r, err := zip.OpenReader(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Close()
+
+	fingerprints := []FileFingerprint{}
+
+	for _, f := range r.File {
+		if filepath.IsAbs(f.Name) || strings.HasPrefix(f.Name, "..") {
+			continue
+		}
+		longFileName := filepath.Join(filename, f.Name) // #nosec
+
+		if !shouldProcessFile(f.FileInfo(), exclusions, longFileName) {
+			continue
+		}
+		rc, err := f.Open()
+		if err != nil {
+			return nil, err
+		}
+		hasher := md5.New()          // #nosec
+		_, err = io.Copy(hasher, rc) // #nosec
```

Comment:
No sec due to ddos and weak crypto, which is fine in this context

Reply:
@zozs opinion on the Copy()?

md5 makes a lot of sense as that is what we use for fingerprinting

Reply:
Copy is fine since the destination stream is MD5, not the filesystem, so zip-bombs won't take space, only time in CI, which only affects the perpetrator themselves :ok: